elastic/apm-server
# modelindexer: improve observability

## Motivation/summary

Improve observability of modelindexer:

- Count number of events rejected due to Elasticsearch queue saturation (`libbeat.output.events.toomany`)
- Count number of events successfully indexed (`libbeat.output.events.acked`)
- Count number of bulk requests (`libbeat.output.events.batches`)
- Add tracing of bulk requests, similar to the libbeat output. Request-level errors are captured.
- Inject trace/transaction ID into logs.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server with `-e -E output.elasticsearch.experimental=true -E instrumentation.enabled=true -E monitoring.enabled=true`
2. Send events to server
3. Check logs for "bulk request completed". This entry should have `trace.id` and `transaciton.id` fields, which should correspond to a trace and transaction with the same IDs.
4. Check stack monitoring docs contain the added metrics

## Related issues

#6360
